basil/TL/Socket.py

```diff
@@ -41,13 +42,18 @@ def write(self, data):
         else:
             cmd = data.encode(self.encoding)
         cmd += self.write_termination.encode(self.encoding)
-        cmd.decode('unicode_escape')
+        if not self.handle_as_byte:
+            cmd.decode('unicode_escape')
         self._sock.send(cmd)
 
     def read(self, buffer_size=1):
         ret = self._sock.recv(buffer_size)
-        data = ret.split(self.read_termination.encode(self.encoding))
-        return data[:-1]
+        if not self.handle_as_byte:
+            data = ret.split(self.read_termination.encode(self.encoding))
+            data = data[:-1]
+        else:
+            data = ret
+        return data
```

Comment:
Simplify to

    ....

    if not self.handle_as_byte:
        return ret.split(self.read_termination.encode(self.encoding))[:-1]
    return ret

Reply:
Done.